tests/metrics_test.go

```diff
@@ -153,14 +157,99 @@ var _ = Describe("Metrics", func() {
 		)
 	})
 
+	Context("SSP metrics", func() {
+		var originalVersion string
+
+		getDeployment := func(name string, namespace string) *apps.Deployment {
+			deployment := &apps.Deployment{}
+			err := apiClient.Get(ctx, client.ObjectKey{Name: name, Namespace: namespace}, deployment)
+			Expect(err).ToNot(HaveOccurred())
+			return deployment
+		}
+
+		updateVersion := func(version string) {
+			Eventually(func() error {
+				deployment := getDeployment(sspDeploymentName, sspDeploymentNamespace)
+
+				envs := deployment.Spec.Template.Spec.Containers[0].Env
+
+				for i, env := range envs {
+					if env.Name == common.OperatorVersionKey {
+						envs[i].Value = version
+						break
+					}
+				}
+
+				return apiClient.Update(ctx, deployment)
+			}, 1*time.Minute, time.Second).ShouldNot(HaveOccurred())
+		}
+
+		waitForVersionUpdate := func(version string) {
+			Eventually(func() error {
+				rsList := &apps.ReplicaSetList{}
+				err := apiClient.List(ctx, rsList, client.InNamespace(sspDeploymentNamespace),
+					client.MatchingLabels{"name": "ssp-operator"})
+				Expect(err).ToNot(HaveOccurred())
+
+				for _, rs := range rsList.Items {
+					for _, env := range rs.Spec.Template.Spec.Containers[0].Env {
+						if env.Name == common.OperatorVersionKey && env.Value == version && rs.Status.ReadyReplicas == *rs.Spec.Replicas {
+							return nil
```

Comment:
It is better to wait until all pods in the ReplicaSet are updated to new version, not only a single pod.

Reply:
Right, sorry, I missed that.